hs_core/hydroshare/resource.py

```diff
@@ -1123,10 +1123,12 @@ def publish_resource(user, pk):
     """
     resource = utils.get_resource_by_shortkey(pk)
 
-    if not resource.can_be_public_or_discoverable:
-        raise ValidationError("This resource cannot be published since it does not have required metadata or content files.")
+    if not resource.can_be_published():
+        raise ValidationError("This resource cannot be published since it does not have required "
+                              "metadata or content files or it is a Web App resource.")
```

Comment:
I would change the message ```or it is a Web App resource``` to something like ```or this resource type is not allowed for publication.``` so that we don't have to hard code the resource type name.